syncer/syncer.go

```diff
@@ -111,15 +116,16 @@ func NewSyncer(ctx context.Context, conf Configuration, ticker layerTicker, mesh
 		syncTimer:         time.NewTicker(conf.SyncInterval),
 		targetSyncedLayer: unsafe.Pointer(&types.LayerID{}),
 		awaitSyncedCh:     make([]chan struct{}, 0),
-		shutdownCtx:       shutdownCtx,
-		cancelFunc:        cancel,
 	}
 }
 
 // Close stops the syncing process and the goroutines syncer spawns.
-func (s *Syncer) Close() {
-	// TODO: ensure goroutines are all terminated before shutting down
-	s.cancelFunc()
+func (s *Syncer) Close() error {
+	s.logger.Info("start terminating syncer goroutines")
+	s.tomb.Kill(errShuttingDown)
+	err := s.tomb.Wait()
```

Comment:
will it exit with errShuttingDown or with nil?

Reply:
errShuttingDown.

Reply:
i think it should be nil or have no error at all. otherwise, caller have to check it and can log it unnecessarily for example